### ISSUE TITLE
M5-04 · Processor UAT & M6 Impact Handoff

### ISSUE THREAD

**Issue by catursetyo** [201dev/cirquo#46]
## Dependencies

M5-01, M5-02, M5-03

## Summary

Validate the complete Processor recovery flow from a routed batch to a terminal
processing outcome. Record Material Flow Ledger evidence, verify measured-weight
reconciliation, and hand the stable event contract to M6 impact dashboards.

## Scope

- Prepare a verified Processor with:
  - accepted material types;
  - supported output types;
  - available daily capacity;
  - valid location and pickup radius.
- Use a Recovery Batch created through the M4 expiry and Circular Routing flow.
- Execute and record these UAT scenarios:
  - the assigned Processor sees an `offered` batch reactively;
  - another Processor cannot read or act on that batch;
  - the assigned Processor accepts a valid, unexpired offer;
  - an expired offer, unsupported material type, and capacity overflow are rejected;
  - the Processor declines an offer and is excluded from future offers;
  - M4 retry receives the declined batch without duplicate routing events;
  - the Processor logs a valid measured intake;
  - invalid, decimal, zero, negative, implausibly high, and future-dated intake
    values are rejected;
  - the Processor logs a valid outcome with output and residual weights;
  - an outcome exceeding accepted weight is rejected;
  - a partial output/residual total records `processLossGrams`;
  - repeated intake and outcome submissions do not duplicate state changes or
    ledger entries;
  - Merchant and Processor views update reactively without manual refresh.
- Inspect the Material Flow Ledger for each tested batch.
- Run `bun run lint`, `bun run build`, and `bun scripts/check-ledger.ts`.
- Write M6 handoff notes for impact aggregation.

## Ledger Evidence Required

For a batch with declared weight `D` and measured intake `M`:

| Event | Delta | Requirement |
|---|---:|---|
| `EXPIRED` from M4 | `-D` | Releases declared unclaimed material |
| `ROUTED` from M4 | `0` | Records Processor offer |
| `INTAKE_DECLINED` when declined | `0` | One event per Processor decline |
| `INTAKE_ACCEPTED` | `+M` | Records Processor-measured intake |
| `PROCESSED` | `-M` | Terminal processing outcome |

`PROCESSED` metadata must contain:

```ts
{
  outputType,
  outputWeightGrams,
  residualWeightGrams,
  processLossGrams,
  conversionRatePercent,
}
```

The conservation check must hold:

```text
LISTED(+D) → EXPIRED(-D) → INTAKE_ACCEPTED(+M) → PROCESSED(-M) = 0
```

Recovered and Residual are derived from `PROCESSED` metadata:

```text
Recovered = outputWeightGrams
Residual = residualWeightGrams
Process loss = acceptedWeightGrams - outputWeightGrams - residualWeightGrams
```

## M6 Handoff Notes

Document the following stable contracts:

- `acceptedWeightGrams` is the Processor’s measured source of truth.
- `offeredWeightGrams` remains immutable as the Merchant-declared routing weight.
- `PROCESSED` is the only event split into Recovered and Residual values.
- M6 must parse `PROCESSED` metadata; it must not treat the full event delta as
  recovered material.
- `processLossGrams` is visible for reconciliation but is not Recovered or Residual.
- `ROUTING_FAILED` contributes residual weight only for batches that M4 could
  not route.
- All M6 metrics must derive from Material Flow Ledger entries, never stored
  dashboard counters or mock data.

## Acceptance Criteria

- [ ] All Processor P0 flows are complete or have accepted linked blocker issues.
- [ ] A verified Processor can complete `offered → accepted → collected → processed`.
- [ ] Server-side ownership, verification, TTL, material-type, and capacity checks work.
- [ ] Measured intake and outcome validation prevent impossible weights.
- [ ] Duplicate intake/outcome submissions create no duplicate ledger events.
- [ ] Ledger entries reconcile for declared and measured weights that differ.
- [ ] Recovered, Residual, and process loss are distinguishable from one another.
- [ ] Merchant and Processor status views update reactively.
- [ ] Mobile queue and forms remain usable at 375px width.
- [ ] UAT evidence contains date, environment, test batch IDs, results, and
  screenshots/recordings without secrets or private credentials.
- [ ] `bun run lint`, `bun run build`, and `bun scripts/check-ledger.ts` pass.
- [ ] M6 handoff notes are attached to the issue.

## Out of Scope

- Building Processor or platform impact dashboards — M6.
- Admin review of intake variance, moderation, and manual re-routing — M7.
- Delivery, dispatch, route optimisation, or Processor logistics tracking.
- Production hardware integrations, offline sync, and connected scales.
